code/modules/reagents/reagents/addictions.dm

```diff
@@ -0,0 +1,47 @@
+#define TICK_ADDICTION_REMINDER 50
+
+/datum/addiction
+	var/name = "addiction"
+	var/tick //Tick goes up if you're not satiating your addiction, Tick goes down if you're satiating it
+	var/addiction = "memes" //Holds the ID for whatever reagent you're addicted to
+	var/mob/holder
+	var/addiction_multiplier = 1
+	var/list/addiction_effects = list() //What effects this addiction will have on the holder. Organized as list(tick_to_trigger = effect)
+
+/datum/addiction/New(var/mob/new_holder, var/addicted)
+	..()
+	if(new_holder)
+		holder = new_holder
+	if(addicted)
+		addiction = addicted
+
+/datum/addiction/Destroy()
+	if(holder)
+		holder = null
+	if(addiction_effects)
+		for(var/datum/addiction_effect/E in addiction_effects)
+			addiction_effects.Remove(E)
+			qdel(E)
+	..()
+
+/datum/addiction/proc/handle_addiction()
+	if(holder.reagents && holder.reagents.has_reagent(addiction, 1*addiction_multiplier))
+		tick = max(0, tick--)
+		if(prob(30))
+			addiction_multiplier+=rand(1,3)/rand(10,30)
+		return
+	tick++
+	handle_withdrawal()
+
+/datum/addiction/proc/handle_withdrawal()
+	if(tick > TICK_ADDICTION_REMINDER)
+		to_chat(holder, "<span class = 'warning'>You need more [addiction]!</span>")
```

Comment:
Should unconscious spessmen get this too?

Reply:
does this not handle for unconsciousness?

Reply:
where?

Reply:
where???

Reply:
Apparently to_chat(thing doesn't check for unconsciousness

Reply:
correct

Reply:
Why would unconscious spessmen not be aware of their own addictions?

Reply:
they shouldn't be, which is what I originally intended to say

Reply:
Never been woken up by a craving?

Reply:
can't say I have

Reply:
Even so, they should then get the message when they actually wake up. Unconscious people can't think, so they have no way to know they need more stuff.